src/service/chat.js

```diff
@@ -258,6 +258,19 @@ const getAllowAnonDmStatus = async (type, id) => {
   }
 };
 
+const checkValidUrl = async (url) => {
+  try {
+    const response = await api.get(url);
+    if (response?.status === 200) {
+      return Promise.resolve(response.data);
+    }
+    return Promise.reject(response.data);
+  } catch (e) {
+    if (e?.response?.data?.message) return Promise.reject(e?.response?.data?.message);
+    return Promise.reject(e);
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `checkValidUrl` function is making a GET request to the provided URL to check if it's valid. This approach can lead to performance issues as it involves network calls which are expensive operations. It also poses a security risk as it opens up possibilities for SSRF (Server Side Request Forgery) attacks where an attacker could potentially provide URLs of internal resources.

A better approach would be to use a regular expression or a library like `validator.js` to validate the URL format without making a network call. Here's how you could do it using a regex:

```diff
- const checkValidUrl = async (url) => {
-   try {
-     const response = await api.get(url);
-     if (response?.status === 200) {
-       return Promise.resolve(response.data);
-     }
-     return Promise.reject(response.data);
-   } catch (e) {
-     if (e?.response?.data?.message) return Promise.reject(e?.response?.data?.message);
-     return Promise.reject(e);
-   }
- };

+ const checkValidUrl = (url) => {
+   const urlRegex = /^(ftp|http|https):\/\/[^ "]+$/;
+   return urlRegex.test(url);
+ };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->